### PR TITLE
chore: upgrade immutable.js and include flow types

### PR DESCRIPTION
**Pull request by drichard-box**
ran this tool to just auto add flowFixMes.
https://github.com/facebook/flow/blob/master/packages/flow-dev-tools/bin/tool

`./bin/tool add-comments --comment "\$FlowFixMe immutable"`

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @drichard-box has signed the CLA. Thanks for the pull request!

File: package.json [hunk 1]
```diff
@@ -192,7 +192,7 @@
         "flow-copy-source": "^2.0.2",
         "form-serialize": "^0.7.2",
         "husky": "^1.1.2",
-        "immutable": "^3.7.4",
+        "immutable": "^3.8.1",
```
Comment (jstoffan) [inline package.json]: Please update the corresponding entry in `peerDependencies` and add `BREAKING CHANGE` to the commit message, as well.

**Comment by jstoffan** [comment]
@drichard-box, can you provide an update on this PR? Is it still needed or can it be closed?

**Comment by jstoffan** [comment]
Closing due to inactivity. Please reopen if this change is still necessary.

**Pull request closed without merging**